fbpic/particles/particles.py

```
@@ -495,6 +511,61 @@ def make_ionizable(self, element, target_species,
             self.int_sorting_buffer = np.empty( self.Ntot, dtype=np.uint64 )
 
+    def activate_spin_tracking(self, sx_m=0., sy_m=0., sz_m=1.,
+                               anom=0.00115965218128, spin_distr='fixed'):
+        """
+        Activate spin tracking for this particle. This will
+        enable calculating the evolution of particle spin
+        vector throughout the simulation.
+
+        Note: spin tracking should be activated _before_ making
+        a species ionizable. Otherwise, spin tracking must be
+        manually activated for the target species of the
+        ionized electrons.
+
+        Parameters
+        ----------
+        sx_m: float
+            The species-averaged average projection onto
+            the x axis
+
+        sy_m: float
+            The species-averaged average projection onto
+            the y axis
+
+        sz_m: float
+            The species-averaged average projection onto
+            the z axis
+
+        anom: float
+            The anomalous magnetic moment of the particle.
+            Default value is that of an electron.
+
+        spin_distr: str, optional
+            If 'fixed', all particles will have a fixed spin value
+            equal to s{x,y,z}_m.
+            If 'rand', the spin vectors will be random, but with an
+            ensemble average defined by one of the values of
+            s{x,y,z}_m. The first non-zero mean component will be
+            used, with order of preference being x,y,z, ie if sx_m!=0,
+            the generated spins will have an ensemble averages of
+            |sx|=sx_m, |sy|=0, |sz|=0, or if sz_m!=0, |sx|=0, |sy|=0
+            and |sz|=sz_m.
+        """
+        # Warn about ionizer!
+        if self.ionizer is not None:
+            warnings.warn('\nIonizer already activated! Please make sure '
+                          'spin tracking is manually activated for all '
+                          'target species\n')
```

See my comment in ionizer.py: I think that it might be cleaner to always explicitly ask the user to manually activate spin tracking, before activating ionization (and raise an error in ionizer.py otherwise).

We could also raise an error here, saying that spin tracking should be activated **before** the ionizer.
```suggestion
```